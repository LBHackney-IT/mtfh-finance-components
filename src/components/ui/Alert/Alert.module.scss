@import 'src/theme';

.container {
  display: flex;
  align-items: center;
  width: 100%;
  height: 50px;
  padding-left: 30px;
  background-color: rgb(206 220 0 / 10%);
  column-gap: 30px;
}

.success {
  border-left: 6px solid lbh-colour('lbh-a03');
  background-color: rgb(164 214 94 / 20%);
}

.info {
  border-left: 6px solid lbh-colour('lbh-a03');
}

.error {
  border-left: 6px solid lbh-colour('lbh-f02');

  h3 {
    color: lbh-colour('lbh-f02');
  }
}

.text {
  line-height: 30px;
}
