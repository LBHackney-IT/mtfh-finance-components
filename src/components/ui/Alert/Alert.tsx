import { memo } from 'react';
import type { PropsWithChildren } from 'react';

import classnames from 'classnames';

import Check from '../../../assets/check.svg';
import HelpOutlined from '../../../assets/help-outlined.svg';
import ErrorTriangle from '../../../assets/error-triangle.svg';

import Typography from '../Typography';
import styles from './Alert.module.scss';

const icons = {
  success: <Check />,
  info: <HelpOutlined />,
  error: <ErrorTriangle />,
};

type AlertSeverity = 'success' | 'info' | 'error';

type AlertProps = PropsWithChildren<{
  className?: string;
  severity: AlertSeverity;
}>;

const Alert = ({ children, severity, className = '' }: AlertProps) => (
  <div className={classnames(styles.container, styles[severity], className)}>
    {icons[severity]}

    <Typography variant="h3" weight={600} className={styles.text}>
      {children}
    </Typography>
  </div>
);

export default memo(Alert);
