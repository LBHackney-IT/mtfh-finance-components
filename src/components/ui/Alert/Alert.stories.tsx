import type { ComponentMeta, ComponentStory } from '@storybook/react';
import Alert from './Alert';

export default {
  title: 'Ui/Alert',
  component: Alert,
  argTypes: {},
} as ComponentMeta<typeof Alert>;

const Template: ComponentStory<typeof Alert> = (args) => <Alert {...args} >Alert</Alert>;

export const Success = Template.bind({});
Success.args = {
  severity: "success"
};

export const Info = Template.bind({});
Info.args = {
  severity: "info"
};

export const Error = Template.bind({});
Error.args = {
  severity: "error"
};
