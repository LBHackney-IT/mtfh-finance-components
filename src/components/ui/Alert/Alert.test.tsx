import { render, screen } from '@testing-library/react';

import Alert from './index';

describe('Alert', () => {
  it('renders with given text', () => {
    render(<Alert severity="success">Notification</Alert>);

    const heading = screen.getByRole('heading', { name: /notification/i });

    expect(heading).toBeInTheDocument();
  });

  it('renders success alert', () => {
    const { container } = render(<Alert severity="success">Alert</Alert>);

    expect(container).toMatchSnapshot();
  });

  it('renders info alert', () => {
    const { container } = render(<Alert severity="info">Alert</Alert>);

    expect(container).toMatchSnapshot();
  });

  it('renders error alert', () => {
    const { container } = render(<Alert severity="error">Alert</Alert>);

    expect(container).toMatchSnapshot();
  });
});
