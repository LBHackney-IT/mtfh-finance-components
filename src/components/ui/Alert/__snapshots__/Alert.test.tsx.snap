// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert renders error alert 1`] = `
<div>
  <div
    class="container info"
  >
    <div />
    <h3
      class="lbh-heading-h3 w600 text"
    >
      Alert
    </h3>
  </div>
</div>
`;

exports[`Alert renders info alert 1`] = `
<div>
  <div
    class="container info"
  >
    <div />
    <h3
      class="lbh-heading-h3 w600 text"
    >
      Alert
    </h3>
  </div>
</div>
`;

exports[`Alert renders success alert 1`] = `
<div>
  <div
    class="container success"
  >
    <div />
    <h3
      class="lbh-heading-h3 w600 text"
    >
      Alert
    </h3>
  </div>
</div>
`;
